lib/s3middleware/azureHelpers/mpuUtils.js

```diff
@@ -172,18 +173,27 @@ dataStoreName, log, cb) => {
                 `Error returned from Azure: ${err}`));
         }
         const numberSubParts = results.length;
+        // check if we have streamed more parts than calculated; should not
+        // occur, but do a sanity assertion to detect any coding logic error
+        if (numberSubParts !== subPartInfo.expectedNumberSubParts) {
```

Comment:
if we're going to do the assertion, we should just do it. No need for the conditional check and the log line. If we crash it will be in the logs.

Reply:
updated so it's just the assertion now